presto-spi/src/main/java/com/facebook/presto/spi/block/SingleRowBlockWriter.java

```diff
@@ -43,6 +44,18 @@
         this.initialBlockBuilderSize = initialBlockBuilderSize;
     }
 
+    // Field block builder can only be obtained before any value has been appended.
+    // Once obtained, normal row-based appending is disabled.
+    // Please make sure each field is appended EXACTLY once.
+    public BlockBuilder getFieldBlockBuilder(int fieldIndex)
+    {
+        if (currentFieldIndexToWrite != 0) {
+            throw new IllegalStateException("column write mode can only be enabled before any value has been appended");
```

Comment:
Update all error messages in this file now that "column/field write mode" is no longer something users of this class explicitly sees.